tox.ini

```diff
@@ -0,0 +1,34 @@
+# Tox (https://tox.readthedocs.io/) is a tool for running tests in multiple virtualenvs. This configuration file helps
+# to run the test suite against different combinations of libraries and Python versions.
+# To use it locally, "pip install tox" and then run "tox --skip-missing-interpreters" from this directory.
+
+[tox]
+isolated_build = true
+minversion = 3.24.3
+envlist = py{37,38,39,310,311,312}
+
+[gh-actions]
+# Mapping of Python versions (MAJOR.MINOR) to Tox factors.
+# When running Tox inside GitHub Actions, the `tox-gh-actions` plugin automatically:
+# 1. Identifies the Python version used to run Tox.
+# 2. Determines the corresponding Tox factor for that Python version, based on the `python` mapping below.
+# 3. Narrows down the Tox `envlist` to environments that match the factor.
+# For more details, please see the `tox-gh-actions` README [0] and architecture documentation [1].
+# [0] https://github.com/ymyzk/tox-gh-actions/tree/v2.8.1
+# [1] https://github.com/ymyzk/tox-gh-actions/blob/v2.8.1/ARCHITECTURE.md
+python =
+    3.7: py37
+    3.8: py38
+    3.9: py39
+    3.10: py310
+    3.11: py311
+    3.12: py312
+
+[testenv]
+allowlist_externals =
+    pytest
+usedevelop = true
+extras =
+    test
+commands =
+    pytest -vv {posargs}
```

Comment:
tox file was created by the upstream project but I updated it to test python from 3.8 onwards.